File: C++/프로그래머스/0/120842. 2차원으로 만들기/2차원으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<int> num_list, int n) {
    int count = num_list.size() / n;
    vector<vector<int>> answer;
    
    for(int i = 0; i < count; i++){
        vector<int> temp_array;
        for(int j = 0; j < n; j++) {
            temp_array.push_back(num_list[(i * n) + j]);
        }
        answer.push_back(temp_array);
    }
    
    return answer;
}

File: Go/프로그래머스/0/120842. 2차원으로 만들기/2차원으로 만들기.go
func solution(num_list []int, n int) [][]int {
    res := make([][]int, len(num_list) / n)
    
    row := -1
    for i, v := range num_list {
        if i % n == 0 {
            row++
            res[row] = make([]int, 0, n)
        }
        
        res[row] = append(res[row], v)
    }
    
    return res
}